sign1.go

```diff
@@ -87,26 +87,49 @@ func (m *Sign1Message) UnmarshalCBOR(data []byte) error {
 //
 // Reference: https://datatracker.ietf.org/doc/html/rfc8152#section-4.4
 func (m *Sign1Message) Sign(rand io.Reader, external []byte, signer Signer) error {
+	return m.sign(rand, nil, external, signer)
+}
+
+// SignDetached signs a Sign1Message using the provided Signer.
```

Comment:
The comments here are very identical to the comments for normal Sign Method.

At the least you should refer to the 
// COSE defined detached payloads in Section 2 of [[RFC9052](https://www.ietf.org/id/draft-steele-cose-hash-envelope-01.html#RFC9052)], defining the Detached payload!

Reply:
> The comments here are very identical to the comments for normal Sign Method.

I'd be surprising if they weren't ;-)

> At the least you should refer to the // COSE defined detached payloads in Section 2 of [[RFC9052](https://www.ietf.org/id/draft-steele-cose-hash-envelope-01.html#RFC9052)]

I reckon the reference to S4.4 of RFC8152 is precise enough.  A standards nerd would probably reference S4.4 of RFC9052 :-), but for the detached payload case, there is no material difference between 9052 and 8152.

Reply:
@thomas-fossati I beg to differ. Section 4.4 RFC8152  talks about general signing. There is no reference to Detached Singing in 4.4

My intention was also to reference to specification which clarifies aspects of DetachedSigning, which is lacking in the reference!

Reply:
> @thomas-fossati I beg to differ. Section 4.4 RFC8152 talks about general signing. There is no reference to Detached Signing in 4.4

And rightly so, because there's nothing special about detached/non-detached in terms of signing and verification.

S4.4, Bullet point 5, second sentence:

> [T]he full payload is used here, _independent of how it is transported._

Reply:
Any decision here?

Reply:
This thread seems resolvable... @shizhMSFT can we resolve this?

Reply:
I'm resolving this as sign is not related to detached or attached, bringing focus to the requested changes.
If you feel otherwise, please activate with specific asks.